packages/bee-q/src/components/option-group/bq-option-group.tsx

```diff
@@ -0,0 +1,81 @@
+import { h, Component, Host, Element } from '@stencil/core';
+
+/**
+ * @part label - The `legend` tag element which acts as a container for the label
+ * @part prefix - The prefix of the label
+ * @part label - The text of the label
+ * @part suffix - The suffix of the label
+ * @part group - The `div` element which holds the option items
+ */
+@Component({
+  tag: 'bq-option-group',
+  styleUrl: './scss/bq-option-group.scss',
+  shadow: true,
+})
+export class BqOptionGroup {
+  // Own Properties
+  // ====================
+
+  // Reference to host HTML element
+  // ===================================
+
+  @Element() el!: HTMLElement;
+
+  // State() variables
+  // Inlined decorator, alphabetical order
+  // =======================================
+
+  // Public Property API
+  // ========================
+
+  // Prop lifecycle events
+  // =======================
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  // Listeners
+  // ==============
+
+  // Public methods API
+  // These methods are exposed on the host element.
+  // Always use two lines.
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  // render() function
+  // Always the last one in the class.
+  // ===================================
+
+  render() {
+    return (
+      <Host>
+        <legend class="option-group__label" part="label">
+          <span class="option-group__label__child" part="prefix">
+            <slot name="header-prefix" />
+          </span>
+          <span class="option-group__label__child label-text" part="label">
+            <slot name="header-label" />
+          </span>
+          <span class="option-group__label__child suffix" part="suffix">
+            <slot name="header-suffix" />
+          </span>
+        </legend>
+        <div class="option-group__container" role="group" aria-label="Options" part="group">
+          <slot />
+        </div>
```

Comment:
Shouldn't we follow the same pattern we have in other components for class names? E.g:

```jsx
<Host>
  <legend class="bq-option-group" part="label">
    <span class="bq-option-group--prefix" part="prefix">
      <slot name="header-prefix" />
    </span>
    <span class="bq-option-group--label" part="label">
      <slot name="header-label" />
    </span>
    <span class="bq-option-group--suffix" part="suffix">
      <slot name="header-suffix" />
    </span>
  </legend>
  <div class="bq-option-group--container" role="group" aria-label="Options" part="group">
    <slot />
  </div>
</Host>
```

There's no need of adding an extra CSS class 👇🏼  to be the one to use for targeting the element, instead of

```jsx
<span class="option-group__label__child label-text" part="label">
  <slot name="header-label" />
</span>
```

do

```jsx
<span class="bq-option-group--label" part="label">
  <slot name="header-label" />
</span>
```
```scss
.bq-option-group--label {
  @apply inline-block w-auto overflow-hidden text-ellipsis whitespace-nowrap;
  @apply pe-[--bq-option-group--label-text-padding-end] ps-[--bq-option-group--label-text-padding-start];
}
```

Remember the `K.I.S.S` principle, Keep It Simple Stupid 🙂